pkg/processor/reportprocessor.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package processor
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/open-cluster-management/insights-client/pkg/config"
+	"github.com/open-cluster-management/insights-client/pkg/types"
+	"github.com/open-cluster-management/insights-client/pkg/retriever"
+	k8sTypes "k8s.io/apimachinery/pkg/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/wg-policy-prototypes/policy-report/api/v1alpha1"
+)
+
+//  patchUint32Value specifies a patch operation for a uint32.
+type patchStringValue struct {
+	Op    string `json:"op"`
+	Path  string `json:"path"`
+	Value string `json:"value"`
+}
+
+// Processor struct
+type Processor struct {
+}
+
+// NewProcessor ...
+func NewProcessor() *Processor {
+	p := &Processor{}
+	return p
+}
+
+// CreateUpdatePolicyReports ...
+func (p *Processor) CreateUpdatePolicyReports(input chan types.ProcessorData) {
+	data := <-input
+	// Loop through Report array and return a PolicyReport for each violation
+	for _, report := range data.Reports.Reports {
+		// Find the correct Insight content data from cache
+		reportData := retriever.ContentsMap[report.Key]
+		if reportData != nil {
```

Comment:
Lets log a message and may be trigger the GET contents to refresh the contentsMap